cpg_pipes/buckets.py

```diff
@@ -0,0 +1,69 @@
+"""
+Utility functions to interact with objects on buckets.
```

Comment:
Should these be part of `cpg_utils`?

Reply:
It probably shouldn't. 
`can_reuse` is something relevant for pipeline-like code - it checks whether stage's outputs can be reused to avoid unnecessary computation.
`exists` checks the object existence, but it also has special cases for `.ht` and `.mt` paths, in which case it would check for the existence of the `_SUCCESS` object. This hail-related special case might not be good for a more generic cpg_utils package.